feat: Add blog guide for posting to our community resources (#321)
open-sauced/docs: docs/contributing/opensauced-writers-guide/how-to-add-your-post.md

---
id: writers-guide
title: "OpenSauced Writers Guide"
sidebar_label: "OpenSauced Writers Guide"
keywords: 
- "writers guide" 
- "open source writers guide" 
- "open source writer" 
---

Welcome to our Writers Contributor Guide. This will walk you through the steps of how to add your blog post to our community resources using the docs repository.

## Repeat Authors: Adding to `authors.yml`

> For regular blog post authors, we should declare those authors globally in the configuration file:

Reviewer: [proofreading]: Could be a bit more clearer
```suggestion
For regular authors, we should declare them globally in the configuration file:
```

Author: I think it's fine the way it is.

**Location:** `/blog/authors.yml`

**Example:**
```yaml
BekahHW:
  name: Bekah Hawrot Weigel
  title: Developer Experience Lead
  url: https://twitter.com/BekahHW
  image_url: https://github.com/BekahHW.png
  email: Bekah@opensauced.pizza

>   twitter: BekahHW

Reviewer: [proofreading/suggestion]: I think it would be helpful to name it x(twitter) for accuracy. 
```suggestion
 x(twitter): BekahHW
```

Author: That's not the docusaurus format. They specifically use Twitter.

Reviewer: Oh ok @BekahHW.

nickytonline:
  name: Nick Taylor

>   title: AI Engingeer

Reviewer: [proofreading]: just a little typo
```suggestion
  title: AI Engineer 
```

  url: https://www.nickyt.co/
  image_url: https://github.com/nickytonline.png

> ```

Reviewer: [suggestion]: for authors who don't use X(Twitter) and use Linkedin, I think it would be helpful to have `linkedin`as an optional value.

Author: It's the docusaurus structure. I don't think you have a choice.

Reviewer: Ahh ok @BekahHW.

## Adding Blog Posts

> To publish in the blog, create a Markdown file within the blog directory. Note that we have subdirectories for each year, so writers should add their post to the folder that corresponds with the year.

Reviewer 2: ```suggestion
To publish in the blog, create a Markdown file within the blog directory. Note that we have subdirectories for each year, so writers should add their posts to the folder that corresponds with the year.
```

**Example Path:** `/blog/2024/06-03-welcome-opensauced.md`

:::info

You'll notice that many of the files include the year. These were created before we started using the year subdirectories. Either way works, but we're moving towards the year subdirectories for organization and omitting the year in the file name.

:::

**Markdown Example:**
```markdown
docs/blog/2024/06-03-welcome-opensauced.md
---
title: Welcome to OpenSauced
description: This is my first post on OpenSauced.

> slug: welcome-opensauced

Reviewer 2: Some folks might not know what slug is. We might want to give more information about this.

Even better, we might want to give all information on how to fill in the metadata. I imagine, folks would ask the same questions, such as: "What should I fill in the slug?" "What tags should I use here?" "What tags are available?" "Should I put my name or username?" and so on.

Author: We actually can't use tags until the docusaurus update gets merged in, but I'll add it in anywhat.

authors: BekahHW
tags: [hello, opensauced]
hide_table_of_contents: false
---

Welcome to this blog. This blog is created with [**OpenSauced**](https://opensauced.pizza).

<!-- truncate -->

This is my first post on OpenSauced.

A whole bunch of exploration to follow.
```

> The front matter is useful to add more metadata to your blog post, for example, author information, but OpenSauced will be able to infer all necessary metadata without the front matter. For all possible fields, see the API documentation.

Reviewer 2: Is there an API docs that we can link here?

> The front matter is useful to add more metadata to your blog post, for example, author information, but OpenSauced will be able to infer all necessary metadata without the front matter. For all possible fields, see the API documentation.

Reviewer 2: Just a nit pick, especially to reduce the commas for better read.

```suggestion
The front matter is useful for adding more metadata to your blog post, such as author information, but OpenSauced will be able to infer all necessary metadata without the front matter. For all possible fields, see the API documentation.
```

### Blog List

The blog's index page (a custom path for OpenSauced at `/community-resources`) is the blog list page, where all blog posts are collectively displayed.

> Use the `<!--truncate-->` marker in your blog post to represent what will be shown as the summary when viewing all published blog posts. Anything above `<!--truncate-->` will be part of the summary. Note that the portion above the truncate marker must be standalone renderable Markdown. For example:

Reviewer 2: >  Note that the portion above the truncate marker must be standalone renderable Markdown. For example:

In my understanding, the "For example:" want to give an example of "standalone renderable Markdown." But on the next line, there is a path example. Unless my understanding is false, I have a couple of suggestions:

- Remove the "For example:" as below:

   ```suggestion
   Use the `<!--truncate-->` marker in your blog post to represent what will be shown as the summary when viewing all published blog posts. Anything above `<!--truncate-->` will be part of the summary. Note that the portion above the truncate marker must be a standalone renderable Markdown.
   ```

OR

- Change the wording a bit. Something like:

   ```suggestion
   Use the `<!--truncate-->` marker in your blog post to represent what will be shown as the summary when viewing all published blog posts. Anything above `<!--truncate-->` will be part of the summary. Note that the portion above the truncate marker must be a standalone renderable Markdown. Take a look at examples below.
   ```

**Example Path:** `website/blog/my-post.md`

**Markdown Example:**
```markdown
website/blog/my-post.md
---
title: Markdown blog truncation example
---

All these will be part of the blog post summary.

<!-- truncate -->

But anything from here on down will not be.
```

### Image Usage

> We are not going to use images in the front matter. The rest of the post should be in markdown format. 

Reviewer 2: There will be no images in the front matter. But can folks add cover image and use images (such as screenshots, screen recordings and gif) for the blog post?

If so, how's the procedure? Like, where to save the images, where to render the cover image, etc.?
